httpx/middleware.py

```diff
@@ -158,3 +149,17 @@ def redirect_content(self, request: AsyncRequest, method: str) -> bytes:
         if request.is_streaming:
             raise RedirectBodyUnavailable()
         return request.content
+
+
+def basic_auth_str(
+    username: typing.Union[str, bytes], password: typing.Union[str, bytes]
+) -> str:
+    if isinstance(username, str):
+        username = username.encode("latin1")
+
+    if isinstance(password, str):
+        password = password.encode("latin1")
+
+    userpass = b":".join((username, password))
+    token = b64encode(userpass).decode().strip()
+    return f"Basic {token}"
```

Comment:
I actually had to extract this helper function for https://github.com/encode/httpx/pull/298 as well. Perhaps worth its own quick PR?